### DISCUSSION TITLE
[@types/aws-lambda] handler.d.ts callback is optional

### DISCUSSION THREAD

**Discussion by bullett445** [DefinitelyTyped/DefinitelyTyped#65579] [Issues with a @types package]
the Handler type should read

export type Handler<TEvent = any, TResult = any> = (
    event: TEvent,
    context: Context,
    callback?: Callback<TResult>,
) => void | Promise<TResult>;

as it is allowed to implement a handler as a async function returning a promise (as stated in the documentation above).

[a comment by an automated account is not shown]

**Comment by bullett445** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/63474

**Comment by bullett445** [comment]
duplicate

**Comment by simonbuchan** [comment]
You... can? The current handler type is:

```ts
export type Handler<TEvent = any, TResult = any> = (
    event: TEvent,
    context: Context,
    callback: Callback<TResult>,
) => null | Promise<TResult>;
```

You can provide an implementation compatible with this in any number of ways, as the existing jsdoc examples on it demonstrate, and the tests in `aws-lambda-tests.ts` verify.

Your only change seems to be making the `callback` optional, as @bullett445 above mentioned this is a long repeated issue that is due to not understanding how typescript checks function signatures: making callback optional is saying that the *caller* can chose to not provide that parameter, it adds the burden to the callee to handle that the parameter might not be provided. Some other existing issues/PRs in this vein:
* #25385
* #30820
* #38342
* #38828
* #55010
* #55105
* #63689
